common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -109,6 +128,27 @@ public static void deleteRecursively(File file) throws IOException {
     }
   }
 
+  private static void deleteRecursivelyUsingUnixNative(File file) throws IOException {
+    final ProcessBuilder builder = new ProcessBuilder("rm", "-rf", file.getAbsolutePath());
+    Process process = null;
+    int exitCode = -1;
+
+    try {
+      process = builder.start();
+      exitCode = process.waitFor();
```

Comment:
@zsxwing : Good catch. I am not going to use Hadoop's ShellCommandExecutor as it will lead to additional dependency for this small thing.